background.ts

```diff
@@ -70,6 +70,17 @@ mainEvents.on('settings-update', (newSettings) => {
   }
 });
 
+// takes care of any propagation of settings we want to do
+// when settings change
+mainEvents.on('settings-update', (newSettings) => {
```

Comment:
Rebase failure — this is (mostly) duplicating a hunk that already landed.  Check lines 65 ~ 71.